AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/resources/ResourceRelationsRest.java

```
@@ -164,11 +153,68 @@ public Response addRelation(@PathParam("slaveResourceGroupName") String slaveGro
+            return Response.status(BAD_REQUEST).entity(new ExceptionDto("Type must either be 'consumed' or 'provided'")).build();
+        }
+
+        Set<? extends AbstractResourceRelationEntity> relations = getRelations(resourceGroupName, releaseName);
+
+        try {
+            removeMatchingRelation(relations, relationName);
+            return Response.status(Response.Status.OK).build();
+        } catch (ForeignableOwnerViolationException fe) {
+            return Response.status(UNAUTHORIZED).entity(new ExceptionDto(fe.getMessage())).build();
+        } catch (ResourceNotFoundException | ElementAlreadyExistsException e) {
+            return Response.status(NOT_FOUND).entity(new ExceptionDto(e.getMessage())).build();
+        }
+    }
+
+    private Set<? extends AbstractResourceRelationEntity> getRelations(String resourceGroupName, String releaseName) throws ValidationException {
+        Set<? extends AbstractResourceRelationEntity> relations = new HashSet<>();
+        if (resourceType.toLowerCase().equals("consumed")) {
+            ResourceEntity resource = resourceLocator.getResourceByNameAndReleaseWithConsumedRelations(resourceGroupName, releaseName);
+            relations = resource.getConsumedMasterRelations();
+        } else if (resourceType.toLowerCase().equals("provided")) {
+            ResourceEntity resource = resourceLocator.getResourceByNameAndReleaseWithProvidedRelations(resourceGroupName, releaseName);
+            relations = resource.getProvidedMasterRelations();
+        }
+        return relations;
+    }
+
+    private void removeMatchingRelation(Collection<? extends AbstractResourceRelationEntity> relations, String relationName)
+            throws ForeignableOwnerViolationException, ResourceNotFoundException, ElementAlreadyExistsException {
+        for (AbstractResourceRelationEntity relation : relations) {
+            if (isMatchingRelationName(relation, relationName)) {

```

What happens if no relations matches? Should return an error.